scripts/test-cache.sh

```
@@ -0,0 +1,487 @@
+PASSED_TESTS=0
+FAILED_TESTS=0
+
+# Parse arguments
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        --json)
+            OUTPUT_MODE="json"
+            ;;
+        --json-dir)
+            shift
+            JSON_DIR="$1"
```

Shellcheck (0.11.0)

[warning] JSON_DIR appears unused. Verify use (or export if used externally).

(SC2034)